keyboards/zsa/moonlander/matrix.c

```diff
@@ -87,115 +64,117 @@ void matrix_init_custom(void) {
-    for (uint8_t row = 0; row <= ROWS_PER_HAND; row++) {
+    for (uint8_t row = 0; row <= MCP_ROWS_PER_HAND; row++) {
         // strobe row
         switch (row) {
-            case 0: gpio_write_pin_high(B10); break;
-            case 1: gpio_write_pin_high(B11); break;
-            case 2: gpio_write_pin_high(B12); break;
-            case 3: gpio_write_pin_high(B13); break;
-            case 4: gpio_write_pin_high(B14); break;
-            case 5: gpio_write_pin_high(B15); break;
-            case 6: break; // Left hand has 6 rows
+            case 0:
+                gpio_write_pin_high(B10);
+                break;
+            case 1:
+                gpio_write_pin_high(B11);
+                break;
+            case 2:
+                gpio_write_pin_high(B12);
+                break;
+            case 3:
+                gpio_write_pin_high(B13);
+                break;
+            case 4:
+                gpio_write_pin_high(B14);
+                break;
+            case 5:
+                gpio_write_pin_high(B15);
+                break;
+            case 6:
+                break; // Left hand has 6 rows
         }
 
-        // right side
-        if (mcp23018_initd) {
-            // #define MCP23_ROW_PINS { GPB5, GBP4, GBP3, GBP2, GBP1, GBP0 }       outputs
-            // #define MCP23_COL_PINS { GPA0, GBA1, GBA2, GBA3, GBA4, GBA5, GBA6 } inputs
-
+        // Selecting the row on the right side of the keyboard.
+        if (!mcp23018_errors) {
             // select row
-            mcp23018_tx[0] = 0x12;                                                                   // GPIOA
-            mcp23018_tx[1] = (0b01111111 & ~(1 << (row))) | ((uint8_t)!mcp23018_leds[2] << 7);       // activate row
-            mcp23018_tx[2] = ((uint8_t)!mcp23018_leds[1] << 6) | ((uint8_t)!mcp23018_leds[0] << 7);  // activate row
-
-            if (MSG_OK != i2c_transmit(MCP23018_DEFAULT_ADDRESS << 1, mcp23018_tx, 3, MOONLANDER_I2C_TIMEOUT)) {
-                dprintf("error hori\n");
-                mcp23018_initd = false;
-            }
-
-            // read col
-
-            mcp23018_tx[0] = 0x13;  // GPIOB
-            if (MSG_OK != i2c_read_register(MCP23018_DEFAULT_ADDRESS << 1, mcp23018_tx[0], &mcp23018_rx[0], 1, MOONLANDER_I2C_TIMEOUT)) {
-                dprintf("error vert\n");
-                mcp23018_initd = false;
-            }
-
-            data = ~(mcp23018_rx[0] & 0b00111111);
-            // data = 0x01;
-        } else {
-            data = 0;
-        }
-
-        if (raw_matrix_right[row] != data) {
-            raw_matrix_right[row] = data;
-            changed         = true;
+            mcp23018_errors += !mcp23018_set_output(MCP23018_DEFAULT_ADDRESS, mcp23018_PORTA, (0b01111111 & ~(1 << (row))) | ((uint8_t)!mcp23018_leds[2] << 7));
+            mcp23018_errors += !mcp23018_set_output(MCP23018_DEFAULT_ADDRESS, mcp23018_PORTB, ((uint8_t)!mcp23018_leds[1] << 6) | ((uint8_t)!mcp23018_leds[0] << 7));
```

Comment:
Probably want to use `mcp23018_set_output_all` here for a small speed boost?

Reply:
any speed boost is good.